### ISSUE TITLE
Compress.exe has stopped working

### ISSUE THREAD

**Issue by csharporbflat** [TestRunnerSRL/OoT-Randomizer#100]
regarless of seed or settings using version .17.1, compress.exe crashes during the process it gets up to the point where it says compressing rom, then within about 1 or 2 seconds compress.exe crashes, 

with the following windows error

Problem signature:
  Problem Event Name:	APPCRASH
  Application Name:	Compress.exe
  Application Version:	0.0.0.0
  Application Timestamp:	5b730262
  Fault Module Name:	StackHash_e0f3
  Fault Module Version:	6.1.7601.24214
  Fault Module Timestamp:	5b6273e1
  Exception Code:	c0000374
  Exception Offset:	00000000000bf6b2
  OS Version:	6.1.7601.2.1.0.256.49
  Locale ID:	1033
  Additional Information 1:	e0f3
  Additional Information 2:	e0f3ddca13ec80444a6753b7377218ce
  Additional Information 3:	e368
  Additional Information 4:	e368e280a53e370decb23474bc8aabd3

the only option is to close it, and then it says rom generated, but the file size is 0. Uncompressed works just fine, but of course you cant use and C button items or it will crash the emulator, so basically unplayable.

OS is win7-x64, no A/V running

**Comment by flagrama** [comment]
Is it actually crashing, or just failing to respond and your force killing it? The latter is pretty common and you just have to wait for it to finish its business.

**Comment by CMuncey** [comment]
Apparently that error code is a heap corruption.. How much memory does your computer have?

**Comment by csharporblfat** [comment]
that particular machine has 8G, it is and X-64bit, so it can use it all. So not a memory issue, as I stated previously, Ampros version 2, compress ran perfectly on the same machine.

**Comment by ChrisMuncey** [comment]
And those are working on the same ROM? Like, you put a rom through the compressor on AA's repo, and then tried the same rom through this one?

If so, then the only other thing I can think of is that I used a different compiler to make them, and somehow that's breaking things.. it could also be something else on your computer interfering with it, though I'm not sure what, and I'm not sure why it would only affect that version..

**Comment by TopHatMan** [comment]
I have no issue with both AA and TEST so the problem lies with this guys hardware and software.

**Comment by csharporblfat** [comment]
ok, so AA version worked on the same machine( let me rephrase AA verision of compress worked with AA versions of randomizer). So how can it be a hardware issue, if AA versions worked on same machine. Only other thing that is different is that it is a VM, but I don't see how that would effect it VM has full 8GB of a 32GB host, and 1 core of a Gen7 intel processor to work with. WIth AA version it was compressing them without errors or problems in about 10 seconds or so. So don't see how it is a hardware version.  

@ChrisMuncey , yes I used the same input rom on both Repos.  Is it possible to have Tests version run through the old compiler for compress so that it should in theory work then with Tests version.

**Comment by CMuncey** [comment]
Wait, so only the VM is messing up? And the VM has 1 core? That's the problem if so. The current version of the compressor doesn't work with 1 core machines due to an oversight. The one I sent you in discord should work.

**Comment by csharporblfat** [comment]
new info -

just for fun I downloaded the compress.exe from AA archive, and used that with Tests version, and it worked. 

So for now that works as a temporary solution.

But I would suggest keeping the bug open so that it may be corrected in future releases.

**Comment by csharporblfat** [comment]
@CMuncey That is usefull information to know, when I get a chance I will try giving the VM another core so it will have 2 available and see if that changes anything.

**Comment by CMuncey** [comment]
The one I uploaded to my fork a few minutes ago (there's a pull request on AA's repo you can steal it from) should work even with 1 core I think.

**Comment by csharporbflat** [comment]
new version is working
